DOC: Update triage team nomination instructions (#31089)
matplotlib/matplotlib: doc/devel/triage.rst

   slightly different example, add it to the original issue as a comment
   or an edit to the original post.

   Label the closed issue with `status: duplicate <https://github.com/matplotlib/matplotlib/issues?q=sort%3Aupdated-desc+state%3Aopen+label%3A%22status%3A+duplicate%22>`__.

#. Make sure that the title accurately reflects the issue. If you have the
   necessary permissions edit it yourself if it's not clear.

#. Add the relevant labels, such as `Documentation <https://github.com/matplotlib/matplotlib/labels?q=documentation>`__
   when the issue is about documentation, `status: confirmed bug <https://github.com/matplotlib/matplotlib/issues?q=sort%3Aupdated-desc+state%3Aopen+label%3A%22status%3A+confirmed+bug%22>`__
   if it is clearly a bug, `New feature <https://github.com/matplotlib/matplotlib/issues?q=sort%3Aupdated-desc+state%3Aopen+label%3A%22New+feature%22>`__
   if it is a new feature request, etc.

   An additional useful step can be to tag the corresponding module e.g.

>    the "GUI/Qt" label when relevant.

Reviewer: ```suggestion
An additional useful step can be to tag with the relevant "topic: ..." label.
```

Author: Added with a couple of examples.

    Note that good first issues are intended to onboard newcomers with a genuine
    interest in improving Matplotlib, in the hopes that they will continue to
    participate in our development community; therefore, the use of AI tools to
    resolve these issues is not appropriate.

Preparing PRs for review
========================

Doing initial reviews of contributions is also encouraged. Contributors and
users are welcome to participate to the review process following our
:ref:`review guidelines <pr-guidelines>`. In particular, if you identify a PR
that needs maintainer attention, you can add the
`status: needs review <https://github.com/matplotlib/matplotlib/issues?q=sort%3Aupdated-desc+state%3Aopen+label%3A%22status%3A+needs+review%22>`_
label to it, or add it to the next community meeting agenda for discussion. You
can:

* Suggest fixes to CI check failures, such as failing tests or documentation
  builds;
* Help with :ref:`rebasing instructions <rewriting-commit-history>`;
* Suggest improvements to the PR description, including filling out the AI
  Disclosure section if it is missing.

AI-generated contributions
--------------------------

> Make sure PRs comply with our :ref:`policy on AI-generated contributions <generative_ai>`.

Reviewer: Simpler:
```suggestion
Make sure PRs comply with our :ref:`AI policy<generative_ai>`.
```

If you identify a PR that does not comply with the policy, ask the contributor
to clarify the AI tools used and the contribution of the author, and to update

> the PR description accordingly.

Reviewer: ```suggestion
to clarify the way AI was used and update to comply with our AI policy.
```

Author: Changed this one slightly, let me know what you think.

.. _triage_team:

Triage team
===========

If you would like to join the triage team:

1. Correctly triage 2-3 issues or review 2-3 pull requests, as described above.
2. Ask someone on in the Matplotlib organization (publicly or privately) to
   recommend you to the triage team (look for "Member" on the top-right of
   comments on GitHub).  If you worked with someone on the issues triaged, they
   would be a good person to ask.
3. Responsibly exercise your new power!

Anyone with commit or triage rights may nominate a user to be invited to join
the triage team by nominating them through the private "Triage team nominations"

> category on `Discourse <https://discourse.matplotlib.org>`__ (Note that only

Reviewer 2: What's the confirmation (this nomination was approved) step?/next part of the process, or is that a follow up PR?

Author: I think this should be a Steering Council action but I wonder if we should add a timeline for a response?

Reviewer 2: I'm not sure it has to be since @qulogic and @ksunden can add people, but agree there should be a timeline. And a confirmation flag like closing the thread?

> ``Staff`` members can see this category).

Reviewer 2: maintainers + triage team members (those are the team categories)

Reviewer 2: https://discourse.matplotlib.org/g/maintainers
https://discourse.matplotlib.org/g/triage

Reviewer 2: And I'd maybe just put the `@s` here

Author: You mean citing people individually?

Reviewer 2: No using the team @ - `@maintainers` `@triage`

> ``Staff`` members can see this category).

Author: ```suggestion
``Staff`` members, belonging to the `@maintainers` or `@triage` groups, can see this category).
```

* Suggest improvements to the PR description, including filling out the AI
  Disclosure section if it is missing.

AI-generated contributions
--------------------------

Make sure PRs comply with our :ref:`AI policy <generative_ai>`. If you identify
a PR that does not comply with the policy, ask the contributor to clarify the AI
tools used and the contribution of the author, and to update the PR description
accordingly to comply with our AI policy.

.. _triage_team:

Triage team
===========

If you would like to join the triage team:

1. Correctly triage 2-3 issues or review 2-3 pull requests, as described above.
2. Ask someone on in the Matplotlib organization (publicly or privately) to
   recommend you to the triage team (look for "Member" on the top-right of
   comments on GitHub).  If you worked with someone on the issues triaged, they
   would be a good person to ask.
3. Responsibly exercise your new power!

Anyone with commit or triage rights may nominate a user to be invited to join
the triage team by nominating them through the private "Triage team nominations"
category on `Discourse <https://discourse.matplotlib.org>`__ (Note that only

> ``maintainers`` and ``triage`` members can see this category). The nomination

Author: ```suggestion
category on `Discourse <https://discourse.matplotlib.org>`__ (Note that only
``@maintainers`` and ``@triage`` members can see this category). The nomination
```

> category on `Discourse <https://discourse.matplotlib.org>`__ (Note that only
> ``@maintainers`` and ``@triage`` members can see this category). The nomination
> will then be confirmed by the Steering Council and the user, if accepted, will
> be added to the triage team on GitHub.

Author: ```suggestion
category on `Discourse <https://discourse.matplotlib.org>`__ (Note that only
``@maintainers`` and ``@triage`` members can see this category).

If no objections are raised in the nomination topic after one week, a member with the `owner` role on GitHub will send an invitation email to the nominee following a template. Once the nominee responds affirmatively, they will add the nominee to the Triage group on GitHub, and close the Discourse thread with a confirmation that the nomination was accepted.
```

Author: How does this look?

Reviewer 2: Content is awesome but I think list format is easier for multi-step process:

```rst

If no objections are raised within one week of the nomination, a member with the `owner` role on GitHub will:
1. Send an invitation email to the nominee following a template. 
2. Once the nominee responds affirmatively, they will add the nominee to the Triage group on GitHub
3. Close the Discourse thread with a confirmation that the nomination was accepted (or turned down).
``` 
But also we should maybe do a `If objections were raised...` as a follow up.

Author: ```suggestion
category on `Discourse <https://discourse.matplotlib.org>`__ (Note that only
``@maintainers`` and ``@triage`` members can see this category). The nomination
will then be confirmed by the Steering Council and the user, if accepted, will
be added to the triage team on GitHub.

If no objections are raised within one week of the nomination, a member with the `owner` role on GitHub will:
1. Send an invitation email to the nominee following a template. 
2. Once the nominee responds affirmatively, they will add the nominee to the Triage group on GitHub
3. Close the Discourse thread with a confirmation that the nomination was accepted (or turned down).

If objections are raised, no action will be taken and the nomination can be revisited in the future.
```

At least, that's how I'm used to working: if objections are raised, it is usually due to a lack of data or a desire to observe the contributor for a little longer. In that case, the nomination proposal can be left open and be revisited in the future. Not sure if we want to add a timeline either, since those can be very flexible, i.e. the person can make a flurry of excellent contributions within the month and that might warrant a second look.

Wdyt?

Reviewer 2: Seems good to me - we can always revisit if there's a contentious nomination.

tools used and the contribution of the author, and to update the PR description
accordingly to comply with our AI policy.

.. _triage_team:

Triage team
===========

If you would like to join the triage team:

1. Correctly triage 2-3 issues or review 2-3 pull requests, as described above.
2. Ask someone on in the Matplotlib organization (publicly or privately) to
   recommend you to the triage team (look for "Member" on the top-right of
   comments on GitHub).  If you worked with someone on the issues triaged, they
   would be a good person to ask.
3. Responsibly exercise your new power!

Anyone with commit or triage rights may nominate a user to be invited to join
the triage team by nominating them through the private "Triage team nominations"
category on `Discourse <https://discourse.matplotlib.org>`__ (Note that only
``@maintainers`` and ``@triage`` members can see this category). The nomination
will then be confirmed by the Steering Council and the user, if accepted, will
be added to the triage team on GitHub.

If no objections are raised within one week of the nomination, a member with the `owner` role on GitHub will:

> 1. Send an invitation email to the nominee following a template. 

Author: ```suggestion
If no objections are raised within one week of the nomination, a member with the ``owner`` role on GitHub will:
1. Send an invitation email to the nominee following a template.
```

> 2. Once the nominee responds affirmatively, they will add the nominee to the Triage group on GitHub

Author: ```suggestion
2. Once the nominee responds affirmatively, they will add the nominee to the Triage group on GitHub, and to the ``@triage`` group on Discourse.
```

Triage team
===========

If you would like to join the triage team:

1. Correctly triage 2-3 issues or review 2-3 pull requests. Actually labeling or
   closing issues and PRs require triage permissions, but you can verify with
   comments if issues have enough information or are still valid, and suggest
   changes to PRs, particularly if you can help contributors solve CI check

>    failures.

Reviewer 2: There's a lot of information here, so wonder if this would work better by being integrated into our how to triage instructions, with a link to those instructions?
